### ISSUE TITLE
My scenario fails at the point where I tap a button, even though the button is successfully tapped

### ISSUE THREAD

**Issue by cjmeyer-polymorph** [jonsamwell/flutter_gherkin#229]
I wrote a scenario where in the middle it says "And I tap the 'continue_sign_in_button' button"

It actually taps the button and loads the next screen, but then it doesn't do anything on the next screen because it says `× And I tap the
'continue_sign_in_button' button # :0 took
104ms
Exception: Bad state: No element`

Can I maybe set it up to continue despite exception?

**Comment by jonsamwell** [comment]
This suggests the button does not have the key `continue_sign_in_button` or it is not present at the time of tap

**Comment by cjmeyer-polymorph** [comment]
Thanks for the help. No, I changed it to the key selection after the "contains "Continue"" did the same thing, and it does have the key. And it does click the button, when I remove that line it doesn't click the button, so it finds it but then fails for no reason.

**Comment by cjmeyer-polymorph** [comment]
```Feature: Sign in
    Using a test user to sign in to the dashboard

    Scenario: Sign in works
    # Background:
        # Given I tap the label that contains the text "Settings"
        # When I tap the button that contains the text "Sign out"
        # And I pause for 1 second
        Given I tap the label that contains the text "Sign In"
        And I fill the 'email_address_input' field with 'capitec@tester.com'
        And I fill the 'password_input' field with 'password_1234'

    Scenario: Continue button
        Given I tap the 'continue_sign_in_button' button
```

**Comment by cjmeyer-polymorph** [comment]
`
Feature: Sign in
    Using a test user to sign in to the dashboard

    Scenario: Sign in works
        Given I tap the label that contains the text "Sign In"
        And I fill the 'email_address_input' field with 'test@test.com'
        And I fill the 'password_input' field with 'password'
        And I tap the 'continue_sign_in_button' button
        Then I expect the widget 'bitcoin_price_widget' to be present within 40 seconds
`

**Comment by jonsamwell** [comment]
Hmmm, interesting can you run it with the logging set to verbose to see what is happening.

https://github.com/jonsamwell/flutter_gherkin/blob/integration_test__package_support/example_with_integration_test/integration_test/gherkin/configuration.dart#L29

It might be worth trying with this branch as it fixes a number of stability errors 

https://github.com/jonsamwell/flutter_gherkin/tree/dart-3.0support

**Comment by WernerRaathAtPolymorph** [comment]
I can confirm the exception being thrown was as follows (using the dart-3.0support branch):
```
Asynchronous call to guarded function leaked.
You must use “await” with all Future-returning test APIs.
The guarded method “pumpAndSettle” from class WidgetTester was
called from
package:flutter_gherkin/src/flutter/adapters/widget_tester_app_driver_adapter.dart
on line 21, but never completed before its parent scope closed.
When the exception was thrown, this was the stack:
#0      TestAsyncUtils.verifyAllScopesClosed (package:flutter_test/src/test_async_utils.dart:291:7)
#1      TestWidgetsFlutterBinding.asyncBarrier (package:flutter_test/src/binding.dart:651:20)
#2      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:834:5)
<asynchronous suspension>
The test description was:
  Sign in works
═════════════════════════════════════════════════════════════════
end of failure 1
```

**Comment by cjmeyer-polymorph** [comment]
Unfortunately still get the same error with   gherkin: ^3.0.0+1 and 
flutter_gherkin:
    git:
      url: https://github.com/jonsamwell/flutter_gherkin.git
      ref: dart-3.0support

**Comment by jonsamwell** [comment]
@cjmeyer-polymorph @WernerRaathAtPolymorph are either of you able to do a reproducing project so I can dig into this? We have tests that cover the `I tap the button that contains the text` step which are passing

**Comment by cjmeyer-polymorph** [comment]
Hi, unfortunately I don't think I'll be able to share the code. But thank you so much for your help, really appreciate it. I see #231 might be a similar issue?

**Comment by jonsamwell** [comment]
Fixed in 3.0,0-rc-11

**Comment by cjmeyer-polymorph** [comment]
Thank you for all your work
